File: WasteVisionWebFE/src/core/domain/ObjectPrediction.ts
import { z } from "zod";

export const ObjectPredictionSchema = z.object({
  id: z.string().optional(),
  confidence: z.string().min(0).max(100),
  category: z.string().min(2).max(100),
  x: z.string().min(0),
  y: z.string().min(0),
  width: z.string().min(0),
  height: z.string().min(0),
  predictionId: z.string(),
});

export type ObjectPrediction = z.infer<typeof ObjectPredictionSchema>;

File: WasteVisionWebBE/Domain/ObjectPrediction/ObjectPrediction.cs
using System;
using DDDSample1.Domain.Shared;
using DDDSample1.Domain.Predictions;

namespace DDDSample1.Domain.ObjectPredictions
{
    public class ObjectPrediction : Entity<ObjectPredictionId>, IAggregateRoot
    {
        public PredictionId PredictionId { get; private set; }
        // Add navigation property
        public Prediction Prediction { get; private set; }
        public NumberDouble X { get; private set; }
        public NumberDouble Y { get; private set; }
        public NumberDouble Width { get; private set; }
        public NumberDouble Height { get; private set; }
        public Description Category { get; private set; }
        public NumberDouble Confidence { get; private set; }

        private ObjectPrediction() { } // For ORM

        public ObjectPrediction(PredictionId predictionId, NumberDouble x, NumberDouble y, 
            Description category, NumberDouble confidence, NumberDouble width, NumberDouble height)
        {
            Id = new ObjectPredictionId(Guid.NewGuid());
            PredictionId = predictionId;
            X = x;
            Y = y;
            Width = width;
            Height = height;
            Category = category;
            Confidence = confidence;
        }
    }
}